### PR TITLE
Make Classmetadata generic

### PR DESCRIPTION
**Pull request by franmomu**
Last bit of https://github.com/doctrine/persistence/pull/159

### CODE REVIEW

**Review by greg0ire — APPROVED** [review]

**Comment by greg0ire** [comment]
@orklah please review

File: tests/Doctrine/Tests/Persistence/Mapping/Fixtures/TestClassMetadata.php [hunk 1]
```diff
@@ -8,25 +8,26 @@
...
     /**
-     * @psalm-param class-string $className
+     * @psalm-param class-string<T> $className
      */
     public function __construct(string $className)
     {
         $this->className = $className;
     }
 
-    /**
```
Comment (orklah) [inline tests/Doctrine/Tests/Persistence/Mapping/Fixtures/TestClassMetadata.php:27]: Why did you remove that?

Comment (franmomu) [reply tests/Doctrine/Tests/Persistence/Mapping/Fixtures/TestClassMetadata.php:27]: Because this class implements `ClassMetadata` and `ClassMetadata::getName()` already has the proper docblock.

https://github.com/doctrine/persistence/pull/171/files#diff-2d04993be2c5f1b2168b04befaa7921df5ec3816fae3cabd7dd084e2686ede43R20

I could add `{@inheritDoc}`.

**Review by orklah — APPROVED** [review]

**Comment by greg0ire** [comment]
Thanks @franmomu !
